File: bin/shell/src/controller/collectionController.py
import json

from src.connection import connection
from src.environmentVariable import EnvironmentVariable
from src.constant import Constant


class CollectionController:

    __SERVICE_NAME = "collection"

    @staticmethod
    def createCollection():
        collectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Collection Name: ")

        print(connection.send(CollectionController.__SERVICE_NAME, "createCollection", [
              EnvironmentVariable.databaseName, collectionName]))

    @staticmethod
    def setCollectionName():
        oldCollectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Old Collection Name: ")

        newCollectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}New Collection Name: ")

        print(connection.send(CollectionController.__SERVICE_NAME, "setCollectionName", [
              EnvironmentVariable.databaseName, oldCollectionName, newCollectionName]))

    @staticmethod
    def getAllObjects():
        collectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Collection Name: ")

        print(connection.send(CollectionController.__SERVICE_NAME, "getAllObjects", [
              EnvironmentVariable.databaseName, collectionName]))

    @staticmethod
    def findFromObjectId():
        collectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Collection Name: ")

        objectId = input(f"{Constant.COMMAND_QUESTION_PREFIX}Object ID: ")

        print(connection.send(CollectionController.__SERVICE_NAME, "findFromObjectId", [
              EnvironmentVariable.databaseName, collectionName, objectId]))

    @staticmethod
    def find():
        collectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Collection Name: ")

        query = json.loads(input(Constant.COMMAND_QUESTION_PREFIX +
                                 "Query (JSON Format like {'db': 'sample_db'}): "))

        print(connection.send(CollectionController.__SERVICE_NAME, "find", [
              EnvironmentVariable.databaseName, collectionName, query]))

    @staticmethod
    def insert():
        collectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Collection Name: ")

        data = json.loads(input(Constant.COMMAND_QUESTION_PREFIX +
                                "Data (JSON Format like {'db': 'sample_db'}): "))

        print(connection.send(CollectionController.__SERVICE_NAME, "insert", [
              EnvironmentVariable.databaseName, collectionName, data]))

    @staticmethod
    def updateFromObjectId():
        collectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Collection Name: ")

        objectId = input(f"{Constant.COMMAND_QUESTION_PREFIX}Object ID: ")

        updateData = json.loads(input(Constant.COMMAND_QUESTION_PREFIX +
                                      "Update Data (JSON Format like {'db': 'sample_db'}): "))

        print(connection.send(CollectionController.__SERVICE_NAME, "updateFromObjectId", [
              EnvironmentVariable.databaseName, collectionName, objectId, updateData]))

    @staticmethod
    def update():
        collectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Collection Name: ")

        query = json.loads(input(Constant.COMMAND_QUESTION_PREFIX +
                                 "Query (JSON Format like {'db': 'sample_db'}): "))

        updateData = json.loads(input(Constant.COMMAND_QUESTION_PREFIX +
                                      "Update Data (JSON Format like {'db': 'sample_db'}): "))

        print(connection.send(CollectionController.__SERVICE_NAME, "update", [
              EnvironmentVariable.databaseName, collectionName, query, updateData]))

    @staticmethod
    def deleteFromObjectId():
        collectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Collection Name: ")

        objectId = input(f"{Constant.COMMAND_QUESTION_PREFIX}Object ID: ")

        print(connection.send(CollectionController.__SERVICE_NAME, "deleteFromObjectId", [
              EnvironmentVariable.databaseName, collectionName, objectId]))

    @staticmethod
    def delete():
        collectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Collection Name: ")

        query = json.loads(input(Constant.COMMAND_QUESTION_PREFIX +
                                 "Query (JSON Format like {'db': 'sample_db'}): "))

        print(connection.send(CollectionController.__SERVICE_NAME, "delete", [
              EnvironmentVariable.databaseName, collectionName, query]))

    @staticmethod
    def drop():
        collectionName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}Collection Name: ")

        print(connection.send(CollectionController.__SERVICE_NAME,
              "drop", [EnvironmentVariable.databaseName, collectionName]))

File: core/source/core/src/main/java/com/padr/tuxdb/server/controller/CollectionController.java
package com.padr.tuxdb.server.controller;

import java.util.Map;

import com.padr.tuxdb.engine.process.pub.Collection;

public class CollectionController {

    @SuppressWarnings("unchecked")
    public static Object handler(String function, Map<String, Object> request) throws Throwable {
        try {
            String databaseName = (String) request.get("databaseName");
            String collectionName = (String) request.get("collectionName");

            Collection collection = new Collection(databaseName, collectionName);

            switch (function) {
                case "createCollection":
                    return Collection.createCollection(databaseName, collectionName);
                case "setCollectionName":
                    return collection.setCollectionName((String) request.get("newCollectionName"));
                case "getAllObjects":
                    return collection.getAllObjects();
                case "findFromObjectId":
                    return collection.findFromObjectId((String) request.get("objectId"));
                case "findOne":
                    return collection.findOne((Map<String, Object>) request.get("query"));
                case "find":
                    return collection.find((Map<String, Object>) request.get("query"));
                case "insert":
                    return collection.insert((Map<String, Object>) request.get("data"));
                case "updateFromObjectId":
                    return collection.updateFromObjectId((String) request.get("objectId"), (Map<String, Object>) request.get("updateData"));
                case "update":
                    return collection.update((Map<String, Object>) request.get("query"), (Map<String, Object>) request.get("updateData"));
                case "deleteFromObjectId":
                    return collection.deleteFromObjectId((String) request.get("objectId"));
                case "delete":
                    return collection.delete((Map<String, Object>) request.get("query"));
                case "drop":
                    return collection.drop();
                case "isExist":
                    return collection.isExist();
                default:
                    return null;
            }
        } catch (Exception e) {
            return null;
        }
    }

}
